src/view/MixerElementSelectorWindow.vala

```diff
@@ -0,0 +1,128 @@
+/*
+ * brickman -- Brick Manager for LEGO MINDSTORMS EV3/ev3dev
+ *
+ * Copyright (C) 2016 Kaelin Laundry <wasabifan@outlook.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * MixerElementSelectorWindow.vala - Lists ALSA mixer elements
+ */
+
+using Ev3devKit;
+using Ev3devKit.Ui;
+
+namespace BrickManager {
+
+    public class MixerElementSelectorWindow : BrickManagerWindow {
+        Ui.Menu element_menu;
+
+        public signal void mixer_elem_selected (IMixerElementViewModel selected_element);
+
+        private class MixerElementWithSignalIds: Object {
```

Comment:
I spent a while trying to figure out how I wanted to do this, but in the end I decided that this was the least worst. This can't be a struct though, because I need to be able to jam it into `represented_object`.